### PR TITLE
Fixed issue on SitePermissionsFeature that doesn't allow its usage on custom tabs

### PR DESCRIPTION
**Pull request by Amejia481**
`SitePermissionsFeature` was using the selected session `onPermissionsResult`, instead of checking the custom session id. This didn't allow its usage on custom tabs. Related issue https://github.com/mozilla-mobile/fenix/issues/2343

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt [hunk 1]
```diff
@@ -213,3 +213,23 @@ fun SessionManager.runWithSession(
     }
     return false
 }
+
+/**
+ * Tries to find a session with the provided session ID or uses them selectedSession and runs the block if found.
+ *
+ * @return True if the session was found and run successfully.
```
Comment (pocmo) [inline components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt]: It says it returns `True` if the session was found and the block run. But it returns what the block returns. So if the block returns `False` then even if the session was found and the block was run it returned false? Maybe that's correct and it's just the wording of this doc that is confusing?

Comment (Amejia481) [reply components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt]: You are right. If the session was found, I was delegating to the lambda function to determine if the block run successfully. But the consumer just wants to know if the block was executed or not. I'll update it to reflect that 👍

File: components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt [hunk 2]
```diff
@@ -222,3 +222,26 @@ fun SessionManager.runWithSession(
...
+        findSessionById(sessionId)?.let { session ->
+            block(session)
+            blockWasRun = true
+        }
+    }
+    selectedSession?.let {
+        block(it)
+        blockWasRun = true
+    }
+    return blockWasRun
```
Comment (pocmo) [inline components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt]: @Amejia481 Wait. Now that you are not returning early.. are you not running the block twice if there's a session id and a selected session?

Comment (Amejia481) [reply components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt]: Fixing it!

Comment (Amejia481) [reply components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt]: Thanks for the catch!

Comment (Amejia481) [reply components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt]: https://github.com/mozilla-mobile/android-components/pull/2970

**Pull request merged**